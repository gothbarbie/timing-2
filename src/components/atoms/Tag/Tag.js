import React from 'react'
import styled from 'styled-components'

const TagStyle = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 2.4rem;
  font-size: 1.4rem;
  font-weight: 500;
  border-radius: 4px;
  border: 1px solid
    ${({ blue, theme }) => (blue ? theme.colors.primary : theme.colors.gray)};
  background-color: ${({ blue, theme }) =>
    blue ? theme.colors.primary : theme.colors.gray};
  color: ${({ theme }) => theme.colors.white};
  outline: none;

  &:focus {
    border: 1px solid
      ${({ blue, theme }) => (blue ? theme.colors.white : theme.colors.primary)};
    background: ${({ blue, theme }) =>
      blue ? theme.colors.primary : theme.colors.primary};
  }

  &:hover {
    border: 1px solid ${({ theme }) => theme.colors.white};
  }

  &:active {
    background: ${({ blue, theme }) =>
      blue ? theme.colors.primary : theme.colors.primary};
    border: 1px solid ${({ theme }) => theme.colors.white};
  }
`

const Tag = ({ blue, name, onClick }) => (
  <TagStyle blue={blue} onClick={onClick}>
    {name}
  </TagStyle>
)

export default Tag
