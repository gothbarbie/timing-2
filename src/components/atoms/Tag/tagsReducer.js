const initialState = [
  {
    name: 'tag',
    count: 1,
  },
]

const tagsReducer = (state = initialState, action) => {
  switch (action.type) {
    default:
      return state
  }
}

export default tagsReducer
