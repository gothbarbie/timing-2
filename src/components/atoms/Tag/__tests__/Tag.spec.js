import React from 'react'
import { shallow } from 'enzyme'

import Tag from '../Tag'

describe('components/atoms/Tag', () => {
  it('renders', () => {
    const component = shallow(<Tag name="Tag 1" onClick={() => jest.fn()} />)

    expect(component).toMatchSnapshot()
  })

  it('renders blue', () => {
    const component = shallow(
      <Tag blue name="Tag 2" onClick={() => jest.fn()} />
    )

    expect(component).toMatchSnapshot()
  })
})
