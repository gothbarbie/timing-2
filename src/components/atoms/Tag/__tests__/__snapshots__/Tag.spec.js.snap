// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/Tag renders 1`] = `
<TagStyle
  onClick={[Function]}
>
  Tag 1
</TagStyle>
`;

exports[`components/atoms/Tag renders blue 1`] = `
<TagStyle
  blue={true}
  onClick={[Function]}
>
  Tag 2
</TagStyle>
`;
